olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -545,95 +469,87 @@ HttpResponse OlpClient::OlpClientImpl::CallApi(
                         "Network request handler is empty.");
   }
 
-  const auto& retry_settings = settings_.retry_settings;
-  auto network_settings =
-      http::NetworkSettings()
-          .WithTransferTimeout(retry_settings.timeout)
-          .WithConnectionTimeout(retry_settings.timeout)
-          .WithProxySettings(
-              settings_.proxy_settings.value_or(http::NetworkProxySettings()));
-
-  auto network_request = http::NetworkRequest(
-      utils::Url::Construct(GetBaseUrl(), path, query_params));
+  auto network_request = CreateRequest(path, method, query_params,
+                                       header_params, post_body, content_type);
 
-  network_request.WithVerb(GetHttpVerb(method))
-      .WithBody(std::move(post_body))
-      .WithSettings(std::move(network_settings));
+  PendingUrlRequestPtr request_ptr = nullptr;
+  auto network = settings_.network_request_handler;
```

Comment:
Do we need to copy it? Can we just use a reference?

Reply:
Also reverted